src/ape/managers/chain.py

```diff
@@ -182,8 +182,11 @@ def range(
         elif start_or_stop < 0:
             raise ValueError(f"stop '{stop}' cannot be negative.")
 
-        for i in range(start, stop, step):
-            yield self._get_block(i)
+        # Note: the range stop block is a non-inclusive stop.
+        #       Where as the query method is an inclusive stop.
+        results = self.query("*", start_block=start, stop_block=stop - 1)  # type: ignore
```

Comment:
Should all of this go in the `_get_block()` helper method on this class so other public methods can make use of this optimization, such as `poll_blocks()`?

Reply:
`poll_blocks` uses range, so we are good:

https://github.com/ApeWorX/ape/blob/main/src/ape/managers/chain.py#L230